src/handshake/client.rs

```diff
@@ -70,9 +71,9 @@ impl<'a, T: AsyncRead + AsyncWrite + Unpin> Client<'a, T> {
 		mem::take(&mut self.buffer)
 	}
 
-	/// Set the handshake origin header.
-	pub fn set_origin(&mut self, o: &'a str) -> &mut Self {
-		self.origin = Some(o);
+	/// Set connection headers to a slice.
+	pub fn set_headers(&mut self, h: &'a [Header]) -> &mut Self {
```

Comment:
should we re-export `httparse::Header` as this is now exposed in the public API?